pkg/timeline/model/repository.ts

```diff
@@ -37,6 +36,12 @@ export interface TimelineRepository {
     noteIDs: readonly NoteID[],
     filter: FetchAccountTimelineFilter,
   ): Promise<Result.Result<Error, Note[]>>;
+
+  /**
+   * @description Fetch list timeline
+   * @param noteId IDs of the notes to be fetched
+   * */
+  fetchListTimeline(noteId: NoteID[]): Promise<Result.Result<Error, Note[]>>;
```

Comment:
```suggestion
  fetchListTimeline(noteId: readonly NoteID[]): Promise<Result.Result<Error, Note[]>>;
```
渡した配列のオブジェクトをレポジトリ側に書き換えされないように, `readonly` を付けたほうがいいと思います.